src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -126,9 +127,21 @@ sealed class ChannelAction {
         data class HtlcInfo(val channelId: ByteVector32, val commitmentNumber: Long, val paymentHash: ByteVector32, val cltvExpiry: CltvExpiry)
         data class StoreHtlcInfos(val htlcs: List<HtlcInfo>) : Storage()
         data class GetHtlcInfos(val revokedCommitTxId: ByteVector32, val commitmentNumber: Long) : Storage()
-        data class StoreIncomingAmount(val amount: MilliSatoshi, val localInputs: Set<OutPoint>, val origin: ChannelOrigin?) : Storage()
-        data class StoreChannelClosing(val amount: MilliSatoshi, val closingAddress: String, val isSentToDefaultAddress: Boolean) : Storage()
-        data class StoreChannelClosed(val closingTxs: List<OutgoingPayment.ClosingTxPart>) : Storage()
+        /** Payment received through on-chain operations (channel creation or splice-in) */
+        sealed class StoreIncomingPayment : Storage() {
+            data class ViaNewChannel(val amount: MilliSatoshi, val serviceFee: MilliSatoshi, val miningFee: Satoshi, val localInputs: Set<OutPoint>, val txId: ByteVector32, val origin: Origin?) : StoreIncomingPayment()
+            data class ViaSpliceIn(val amount: MilliSatoshi, val serviceFee: MilliSatoshi, val miningFee: Satoshi, val localInputs: Set<OutPoint>, val txId: ByteVector32, val origin: Origin.PayToOpenOrigin?) : StoreIncomingPayment()
+        }
+        /** Payment received through on-chain operations (channel close or splice-out) */
+        sealed class StoreOutgoingPayment : Storage() {
+            abstract val amount: Satoshi
+            abstract val miningFees: Satoshi
+            abstract val address: String
+            abstract val txId: ByteVector32
+            data class ViaSpliceOut(override val amount: Satoshi, override val miningFees: Satoshi, override val address: String, override val txId: ByteVector32) : StoreOutgoingPayment()
+            data class ViaClose(override val amount: Satoshi, override val miningFees: Satoshi, override val address: String, override val txId: ByteVector32, val isSentToDefaultAddress: Boolean, val closingType: Type) : StoreOutgoingPayment() { enum class Type { Mutual, Local, Remote, Revoked, Other; } }
+        }
+        data class SetConfirmationStatus(val txId: ByteVector32, val status: ConfirmationStatus) : Storage() { enum class ConfirmationStatus { NOT_LOCKED, LOCKED } }
```

Comment:
The goal of this status is to tell integrators when they can display the payment as "final" to their users. Depending on the nature of the transaction, it may require local confirmations, or local+remote confirmations, or no confirmation at all, hence the awkwardness in naming (mix of "confirmation" and "lock").

Another thing is that we never really use the `NOT_LOCKED` status. We always:
1) emit a `StoreIncomingPayment` or `StoreOutgoingPayment` (containing a `txId`)
2) then emit `SetConfirmationStatus(txId, LOCKED)` (can be immediately after, or take days)

Maybe this should be renamed to `SetFinal`?